adapters/core2p2p/block.go

```diff
@@ -28,29 +27,44 @@ func AdaptSignature(sig []*felt.Felt) *spec.ConsensusSignature {
 	}
 }
 
-func AdaptHeader(header *core.Header, commitments *core.BlockCommitments) *spec.BlockHeader {
-	return &spec.BlockHeader{
+func AdaptHeader(header *core.Header, commitments *core.BlockCommitments) *spec.SignedBlockHeader {
+	return &spec.SignedBlockHeader{
+		BlockHash:        AdaptHash(header.Hash),
 		ParentHash:       AdaptHash(header.ParentHash),
 		Number:           header.Number,
-		Time:             timestamppb.New(time.Unix(int64(header.Timestamp), 0)),
+		Time:             header.Timestamp,
 		SequencerAddress: AdaptAddress(header.SequencerAddress),
-		ProofFact:        nil, // not defined yet
-		Receipts:         nil, // not defined yet
-		StateDiffs:       nil, // not defined yet
-		State: &spec.Patricia{
-			Height: core.ContractStorageTrieHeight,
-			Root:   AdaptHash(header.GlobalStateRoot),
-		},
-		Transactions: &spec.Merkle{
-			NLeaves: uint32(header.TransactionCount),
+		StateRoot:        AdaptHash(header.GlobalStateRoot),
+		Transactions: &spec.Patricia{
+			NLeaves: header.TransactionCount,
 			Root:    AdaptHash(commitments.TransactionCommitment),
 		},
-		Events: &spec.Merkle{
-			NLeaves: uint32(header.EventCount),
+		Events: &spec.Patricia{
+			NLeaves: header.EventCount,
 			Root:    AdaptHash(commitments.EventCommitment),
 		},
+		// todo fill receipts
+		Receipts:        nil,
 		ProtocolVersion: header.ProtocolVersion,
-		GasPrice:        AdaptFelt(header.GasPrice),
+		GasPriceFri:     AdaptUint128(header.GasPrice),
+		Signatures:      utils.Map(header.Signatures, AdaptSignature),
+		// todo(kirill) set these fields
```

Comment:
I thought given existing info from db we can calculate this field (for some reason todo mention several fields, but currently only one is missing - `StateDiffCommitment`

Reply:
yeah, that's what I meant. 

We have 2 options:
- We calculate the state diff commitment and store it in DB when we get it from the feeder by adding a new field to StateDiff (will require a migration).
- Don't store the state diff commitment in the DB and recalculate every time we serve the Statediff. The disadvantage of this is that it may be slow while serving the request since hash calculations are not fast.

I think short-term option 2 is better (since no one really cares about p2p performance right now) longer term we should go for option 1.

Reply:
This can be done in a separate PR

Reply:
sure, will do